src/services/download/gcs.rs

```diff
@@ -273,3 +273,153 @@ impl GcsDownloader {
         .collect()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::sources::{CommonSourceConfig, DirectoryLayoutType, SourceId};
+    use crate::test;
+    use crate::types::ObjectType;
+
+    use super::*;
+    use sha1::{Digest as _, Sha1};
+
+    fn gcs_source_key() -> Option<GcsSourceKey> {
+        let private_key = std::env::var("SENTRY_SYMBOLICATOR_GCS_PRIVATE_KEY").ok()?;
+        let client_email = std::env::var("SENTRY_SYMBOLICATOR_GCS_CLIENT_EMAIL").ok()?;
+
+        if private_key.is_empty() || client_email.is_empty() {
+            None
+        } else {
+            Some(GcsSourceKey {
+                private_key,
+                client_email,
+            })
+        }
+    }
+
+    macro_rules! gcs_source_key {
+        () => {
+            match gcs_source_key() {
+                Some(key) => key,
+                None => {
+                    println!("Skipping due to missing SENTRY_SYMBOLICATOR_GCS_PRIVATE_KEY or SENTRY_SYMBOLICATOR_GCS_CLIENT_EMAIL");
+                    return;
+                }
+            }
+        }
+    }
+
+    fn gcs_source(source_key: GcsSourceKey) -> Arc<GcsSourceConfig> {
+        Arc::new(GcsSourceConfig {
+            id: SourceId::new("gcs-test"),
+            bucket: "sentryio-system-symbols-0".to_owned(),
```

Comment:
Not related to this test, i am somewhat worried about us connecting to a real system bucket for tests.  I hope our credentials are at least read only?  but i suspect they might not be

Reply:
I had the same concerns re depending on external services in the tests. We looked up how to do mocking for the s3 case, and it would basically change the complete code that we test.

Reply:
The credentials are read-only, checked. Thanks.